### PR TITLE
Custom configuration for consumer and producer

### PR DESCRIPTION
**Pull request by kalondar**
Hi

We need to test message flow with messages bigger than default 1MB.
I've prepared a pull request with additional custom producer and consumer configurations.
I also created a unit test for it and updated README with code example.
Can you take a look ?
Any comments welcome

In near future, I plan to merge it to 0.8 branch, because we need it on Spark 1.6

Regards

### CODE REVIEW

**Comment by kalondar** [comment]
Strange, local IDE reports successful tests, but Travis failed.
Sorry for that. Let me investigate a bit

**Comment by kalondar** [comment]
I have no rights to retrigger that build. Can you ?

File: embedded-kafka/src/test/scala/net/manub/embeddedkafka/EmbeddedKafkaCustomConfigSpec.scala [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package net.manub.embeddedkafka
+
+import scala.language.postfixOps
+
+class EmbeddedKafkaCustomConfigSpec extends EmbeddedKafkaSpecSupport with EmbeddedKafka {
+  val TWO_MEGABYTES = 2097152
```
Comment (manub) [inline embedded-kafka/src/test/scala/net/manub/embeddedkafka/EmbeddedKafkaCustomConfigSpec.scala]: these should be probably be named `TwoMegabytes` and `ThreeMegabytes`  as per http://docs.scala-lang.org/style/naming-conventions.html

File: embedded-kafka/src/test/scala/net/manub/embeddedkafka/EmbeddedKafkaCustomConfigSpec.scala [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  val THREE_MEGABYTES = 3145728
+
+  "the custom config" should {
+    "should allow pass additional producer parameters" in {
```
Comment (manub) [inline embedded-kafka/src/test/scala/net/manub/embeddedkafka/EmbeddedKafkaCustomConfigSpec.scala]: `should` is redundant here :)

File: embedded-kafka/src/test/scala/net/manub/embeddedkafka/EmbeddedKafkaCustomConfigSpec.scala [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+      withRunningKafka {
+        publishStringMessageToKafka(topic, bigMessage)
+        consumeFirstStringMessageFrom(topic) shouldBe bigMessage
+      }
+    }
+  }
+
+  def generateMessageOfLength(length: Int): String = Stream.continually(util.Random.nextPrintableChar) take length mkString
+}
```
Comment (manub) [inline embedded-kafka/src/test/scala/net/manub/embeddedkafka/EmbeddedKafkaCustomConfigSpec.scala]: newline please

File: embedded-kafka/src/test/scala/net/manub/embeddedkafka/EmbeddedKafkaCustomConfigSpec.scala (hunk 1 above)
Comment (kalondar) [inline embedded-kafka/src/test/scala/net/manub/embeddedkafka/EmbeddedKafkaCustomConfigSpec.scala]: Good point

File: embedded-kafka/src/test/scala/net/manub/embeddedkafka/EmbeddedKafkaCustomConfigSpec.scala (hunk 2 above)
Comment (kalondar) [inline embedded-kafka/src/test/scala/net/manub/embeddedkafka/EmbeddedKafkaCustomConfigSpec.scala]: Ehhh, yes :)

File: embedded-kafka/src/test/scala/net/manub/embeddedkafka/EmbeddedKafkaCustomConfigSpec.scala (hunk 3 above)
Comment (kalondar) [inline embedded-kafka/src/test/scala/net/manub/embeddedkafka/EmbeddedKafkaCustomConfigSpec.scala]: Fixed

**Comment by kalondar** [comment]
Fixed

**Comment by manub** [comment]
Thanks! You were mentioning you need this on 0.8 branch?

**Comment by kalondar** [comment]
Yes. Now I'm sure you satisfied my code, I can start moving that code to 0.8 branch, and test it there

**Pull request merged**